src/core/lib/surface/call.cc

```diff
@@ -2352,6 +2365,30 @@ void PromiseBasedCall::SetCompletionQueue(grpc_completion_queue* cq) {
       grpc_polling_entity_create_from_pollset(grpc_cq_pollset(cq));
 }
 
+void PromiseBasedCall::UpdateDeadline(Timestamp deadline) {
+  if (deadline >= deadline_) return;
+  auto* const event_engine = channel()->event_engine();
+  if (deadline_ != Timestamp::InfFuture()) {
+    if (!event_engine->Cancel(deadline_task_)) return;
+  } else {
+    InternalRef("deadline");
+  }
+  event_engine->RunAfter(deadline - Timestamp::Now(), this);
```

Comment:
Do we want this to use the cached value of "now"?  Just wondering if there are going to be any issues where this could cause the timer to fire later than it should.

Reply:
I think this should be fine: promise based calls are going to cache just for one poll, and those polls are going to be relatively quick.